src/main/java/datawave/query/util/DateFrequencyValue.java

```diff
@@ -0,0 +1,198 @@
+ * maximum length for a 10 year capture of 14720 bytes.
+ */
+
+public class DateFrequencyValue {
+    
+    private static final Logger log = LoggerFactory.getLogger(DateFrequencyValue.class);
+    
+    public static final int DAYS_IN_LEAP_YEAR = 366;
+    private static int NUM_YEAR_BYTES = 4;
+    private static int NUM_FREQUENCY_BYTES = DAYS_IN_LEAP_YEAR * 4;
+    private static int NUM_BYTES_PER_FREQ_VALUE = 4;
+    
+    public DateFrequencyValue() {}
+    
+    /**
+     * @param dateToFrequencyValueMap
+     *            the keys should be dates in yyyyMMdd format
+     * @return size the size of the ByteArrayOutputStream
+     */
+    private int calculateOutputArraySize(TreeMap<YearMonthDay,Frequency> dateToFrequencyValueMap) {
+        int year, presentYear = 0, size = 0;
+        
+        for (Map.Entry<YearMonthDay,Frequency> dateFrequencyEntry : dateToFrequencyValueMap.entrySet()) {
+            year = dateFrequencyEntry.getKey().year;
+            if (year != presentYear) {
+                size += (NUM_YEAR_BYTES + NUM_FREQUENCY_BYTES);
+                presentYear = year;
+                if (log.isTraceEnabled())
+                    log.trace("Estimating size for year : " + presentYear);
+                log.info("Estimating size for year : " + presentYear);
+            }
+        }
+        
+        return size;
+    }
+    
+    /**
+     *
+     * @param dateToFrequencyValueMap
+     *            the keys should be dates in yyyyMMdd format
+     * @return Value the value to store in accumulo
+     */
+    public Value serialize(TreeMap<YearMonthDay,Frequency> dateToFrequencyValueMap) {
+        
+        Value serializedMap;
+        int year, presentYear = 0;
+        ByteArrayOutputStream baos = new ByteArrayOutputStream(calculateOutputArraySize(dateToFrequencyValueMap));
+        int ordinal, nextOrdinal = 1;
+        OrdinalDayOfYear ordinalDayOfYear = null;
+        
+        for (Map.Entry<YearMonthDay,Frequency> dateFrequencyEntry : dateToFrequencyValueMap.entrySet()) {
+            year = dateFrequencyEntry.getKey().year;
+            ordinal = dateFrequencyEntry.getKey().julian;
+            
+            if (year != presentYear) {
+                
+                if (nextOrdinal > 1 && (nextOrdinal < DAYS_IN_LEAP_YEAR)) {
+                    // Add zero frequencies for the remaining days of the year that were not in the map
+                    // for the last year processed
+                    do {
+                        Base256Compression.writeToOutputStream(0, baos);
+                        nextOrdinal++;
+                    } while (nextOrdinal < (DAYS_IN_LEAP_YEAR + 1));
+                }
+                
+                ordinalDayOfYear = new OrdinalDayOfYear(ordinal, year);
+                nextOrdinal = 1;
+                
+                try {
+                    baos.write(Base256Compression.numToBytes(year));
```

Comment:
You are still allocating 4 byte arrays.  Please do not do this.  Please use your writeToOutputStream(year, baos) method.